src/core/net/nat64_translator.cpp

```diff
@@ -619,6 +640,7 @@ void Translator::SetEnabled(bool aEnabled)
     }
 
     LogInfo("NAT64 translator %s", aEnabled ? "enabled" : "disabled");
+    UpdateState();
```

Comment:
Just an idea: now that we have `mState` I think we may be able to remove `mEnabled` and use `mState` directly?
It is fine to keep it for now (can consider it for later and see if it make sense).

Reply:
After some attempts, I would prefer a separate PR to have a deeper review the changes around the mEnabled since it is not a trival change for me.

Reply:
SGTM.